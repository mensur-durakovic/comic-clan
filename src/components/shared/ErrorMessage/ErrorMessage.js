import React from "react";
import styled from "styled-components";

const ErrorMessageWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: 50vh;
`;

const ErrorMessageText = styled.span`
  color: #f15454;
  font-size: 40px;
  text-align: center;
`;

const ErrorMessage = React.memo((props) => {
  const { text } = props;
  return (
    <ErrorMessageWrapper>
      <ErrorMessageText>{text}</ErrorMessageText>
    </ErrorMessageWrapper>
  );
});

export default ErrorMessage;
