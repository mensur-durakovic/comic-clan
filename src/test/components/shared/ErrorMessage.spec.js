import React from 'react';
import renderer from 'react-test-renderer';
import ErrorMessage from '../../../components/shared/ErrorMessage/ErrorMessage';

test('ErrorMessage component', () => {
  const tree = renderer.create(<ErrorMessage text="test"/>).toJSON()
  expect(tree).toMatchSnapshot()
})