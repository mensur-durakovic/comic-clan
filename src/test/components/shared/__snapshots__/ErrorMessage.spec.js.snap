// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component 1`] = `
<div
  className="sc-AxjAm gEktpb"
>
  <span
    className="sc-AxirZ kLQCpT"
  >
    test
  </span>
</div>
`;
